src/coreclr/src/pal/src/misc/cgroup.cpp

```diff
@@ -300,27 +325,47 @@ class CGroup
                 maxLineLen = lineLen;
             }
 
-            // See man page of proc to get format for /proc/self/cgroup file
-            int sscanfRet = sscanf_s(line,
-                                     "%*[^:]:%[^:]:%s",
-                                     subsystem_list, lineLen+1,
-                                     cgroup_path, lineLen+1);
-            if (sscanfRet != 2)
+            if (s_cgroup_version == 1)
             {
-                _ASSERTE(!"Failed to parse cgroup info file contents with sscanf_s.");
-                goto done;
-            }
+                // See man page of proc to get format for /proc/self/cgroup file
+                int sscanfRet = sscanf_s(line,
+                                         "%*[^:]:%[^:]:%s",
+                                         subsystem_list, lineLen+1,
+                                         cgroup_path, lineLen+1);
+                if (sscanfRet != 2)
+                {
+                    _ASSERTE(!"Failed to parse cgroup info file contents with sscanf.");
+                    goto done;
+                }
 
-            char* context = nullptr;
-            char* strTok = strtok_s(subsystem_list, ",", &context);
-            while (strTok != nullptr)
+                char* context = nullptr;
+                char* strTok = strtok_r(subsystem_list, ",", &context);
```

Comment:
What is the benefit of this change from strtok_s to strtok_r?

Reply:
Fixed now. Sorry, I had the original implementation in the non-pal cgroup.cpp (it can be built and run by itself) and forgot to fix this when I copied it over to PAL.